bug: timeout causes unknown error

```
2020-11-29 13:31:53.576 10548-11382/de.robingenz.dhbw.dualis W/Cordova-Plugin-HTTP: Request timed out
    com.silkimen.http.HttpRequest$HttpRequestException: java.net.SocketTimeoutException: timeout
        at com.silkimen.http.HttpRequest.code(HttpRequest.java:1449)
        at com.silkimen.http.HttpRequest.stream(HttpRequest.java:1740)
        at com.silkimen.http.HttpRequest.buffer(HttpRequest.java:1729)
        at com.silkimen.http.HttpRequest.receive(HttpRequest.java:1856)
        at com.silkimen.cordovahttp.CordovaHttpBase.processResponse(CordovaHttpBase.java:195)
        at com.silkimen.cordovahttp.CordovaHttpBase.run(CordovaHttpBase.java:81)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.net.SocketTimeoutException: timeout
        at com.android.okhttp.okio.Okio$3.newTimeoutException(Okio.java:214)
        at com.android.okhttp.okio.AsyncTimeout.exit(AsyncTimeout.java:263)
        at com.android.okhttp.okio.AsyncTimeout$2.read(AsyncTimeout.java:217)
        at com.android.okhttp.okio.RealBufferedSource.indexOf(RealBufferedSource.java:307)
        at com.android.okhttp.okio.RealBufferedSource.indexOf(RealBufferedSource.java:301)
        at com.android.okhttp.okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:197)
        at com.android.okhttp.internal.http.Http1xStream.readResponse(Http1xStream.java:188)
        at com.android.okhttp.internal.http.Http1xStream.readResponseHeaders(Http1xStream.java:129)
        at com.android.okhttp.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:750)
        at com.android.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:622)
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:475)
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:411)
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:542)
        at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getResponseCode(DelegatingHttpsURLConnection.java:106)
        at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:30)
        at com.silkimen.http.HttpRequest.code(HttpRequest.java:1447)
        at com.silkimen.http.HttpRequest.stream(HttpRequest.java:1740) 
        at com.silkimen.http.HttpRequest.buffer(HttpRequest.java:1729) 
        at com.silkimen.http.HttpRequest.receive(HttpRequest.java:1856) 
        at com.silkimen.cordovahttp.CordovaHttpBase.processResponse(CordovaHttpBase.java:195) 
        at com.silkimen.cordovahttp.CordovaHttpBase.run(CordovaHttpBase.java:81) 
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
     Caused by: java.net.SocketException: socket is closed
        at com.android.org.conscrypt.ConscryptFileDescriptorSocket$SSLInputStream.read(ConscryptFileDescriptorSocket.java:554)
        at com.android.okhttp.okio.Okio$2.read(Okio.java:138)
        at com.android.okhttp.okio.AsyncTimeout$2.read(AsyncTimeout.java:213)
        at com.android.okhttp.okio.RealBufferedSource.indexOf(RealBufferedSource.java:307) 
        at com.android.okhttp.okio.RealBufferedSource.indexOf(RealBufferedSource.java:301) 
        at com.android.okhttp.okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:197) 
        at com.android.okhttp.internal.http.Http1xStream.readResponse(Http1xStream.java:188) 
        at com.android.okhttp.internal.http.Http1xStream.readResponseHeaders(Http1xStream.java:129) 
        at com.android.okhttp.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:750) 
        at com.android.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:622) 
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:475) 
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:411) 
        at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:542) 
        at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getResponseCode(DelegatingHttpsURLConnection.java:106) 
        at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:30) 
        at com.silkimen.http.HttpRequest.code(HttpRequest.java:1447) 
        at com.silkimen.http.HttpRequest.stream(HttpRequest.java:1740) 
        at com.silkimen.http.HttpRequest.buffer(HttpRequest.java:1729) 
        at com.silkimen.http.HttpRequest.receive(HttpRequest.java:1856) 
        at com.silkimen.cordovahttp.CordovaHttpBase.processResponse(CordovaHttpBase.java:195) 
        at com.silkimen.cordovahttp.CordovaHttpBase.run(CordovaHttpBase.java:81) 
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
2020-11-29 13:31:53.586 10548-10548/de.robingenz.dhbw.dualis E/Capacitor/Console: File: http://192.168.2.120:8100/main.js - Line 41 - Msg: [NativeHttpService] [object Object]
2020-11-29 13:31:53.814 10548-10548/de.robingenz.dhbw.dualis E/Capacitor/Console: File: http://192.168.2.120:8100/main.js - Line 135 - Msg: Error: Uncaught (in promise): Error: [NativeHttpService] Unknown error occurred.
    Error: [NativeHttpService] Unknown error occurred.
```

--- a/src/app/core/services/native-http/native-http.service.ts
+++ b/src/app/core/services/native-http/native-http.service.ts
@@ -56,6 +56,9 @@ export class NativeHttpService {
   }
 
   private parseError(error: any): Error {
+    if (error instanceof Object && 'rejection' in error) {
+      error = (error as any).rejection;
+    }
     if (!error.error || !error.status || !error.url) {
       return new Error(`${LOGTAG} Unknown error occurred.`);
     }

